packages/core/admin/admin/src/components/LeftMenu.tsx

```diff
@@ -84,68 +84,123 @@ const LeftMenu = ({ generalSectionLinks, pluginsSectionLinks }: LeftMenuProps) =
   );
   const listLinks = sortLinks(listLinksAlphabeticallySorted);
 
+  /**
+   * Return filtered mobile navigation links (used for both top and burger menu)
+   */
+  const mapMobileNavigationLinks = React.useCallback(
+    (mobileNavLinks: MobileMenuItem[]): MenuItem[] =>
+      mobileNavLinks
+        .map((mobileLink) => {
+          const linkFound = listLinks.find((link) => link.to === mobileLink.to);
+          if (!linkFound) return null;
+          return mobileLink.link ? { ...linkFound, navigationLink: mobileLink.link } : linkFound;
+        })
+        .filter((link) => link !== null) as MenuItem[],
+    [listLinks]
```

Comment:
I don't think the callback does any optimization here. It re-evaluates the value of mapMobileNavigationLinks every time listLinks gets a new identity, which is on every re-render